packages/headless/src/controllers/knowledge/generatedAnswer/headless-knowledge-generated-answer.test.ts

```diff
@@ -0,0 +1,285 @@
+  dislikeGeneratedAnswer,
+  likeGeneratedAnswer,
+  openGeneratedAnswerFeedbackModal,
+  registerFieldsToIncludeInCitations,
+  sendGeneratedAnswerFeedback,
+  setIsVisible,
+  updateResponseFormat,
+} from '../../../features/generated-answer/generated-answer-actions';
+import {
+  generatedAnswerAnalyticsClient,
+  GeneratedAnswerFeedback,
+} from '../../../features/generated-answer/generated-answer-analytics-actions';
+import {
+  GeneratedAnswerState,
+  getGeneratedAnswerInitialState,
+} from '../../../features/generated-answer/generated-answer-state';
+import {
+  buildMockSearchEngine,
+  MockedSearchEngine,
+} from '../../../test/mock-engine-v2';
+import {createMockState} from '../../../test/mock-state';
+import {
+  GeneratedAnswerProps,
+  GeneratedResponseFormat,
+} from '../../generated-answer/headless-generated-answer';
+import {buildKnowledgeGeneratedAnswer} from './headless-knowledge-generated-answer';
+
+jest.mock('../../../features/generated-answer/generated-answer-actions');
+jest.mock(
+  '../../../features/generated-answer/generated-answer-analytics-actions'
+);
+jest.mock('../../../features/search/search-actions');
+
+describe('knowledge-generated-answer', () => {
+  it('should be tested', () => {
+    expect(true).toBe(true);
+  });
+
+  let engine: MockedSearchEngine;
+
+  const createGeneratedAnswer = (props: GeneratedAnswerProps = {}) =>
+    buildKnowledgeGeneratedAnswer(
+      engine,
+      generatedAnswerAnalyticsClient,
+      props
+    );
+
+  const buildEngineWithGeneratedAnswer = (
+    initialState: Partial<GeneratedAnswerState> = {}
+  ) => {
+    const state = createMockState({
+      generatedAnswer: {
+        ...getGeneratedAnswerInitialState(),
+        ...initialState,
+      },
+    });
+    return buildMockSearchEngine(state);
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    engine = buildEngineWithGeneratedAnswer();
+  });
+
+  it('initializes', () => {
+    const generatedAnswer = createGeneratedAnswer();
+    expect(generatedAnswer).toBeTruthy();
+  });
+
+  it('exposes its state', () => {
+    const generatedAnswer = createGeneratedAnswer();
+
+    expect(generatedAnswer.state).toEqual({
+      ...engine.state.generatedAnswer,
+      answer: undefined,
+      answerContentFormat: 'text/plain',
+      error: {message: undefined},
+    });
+  });
+
+  it('dispatches a rephrase action', () => {
+    const generatedAnswer = createGeneratedAnswer();
+    const responseFormat: GeneratedResponseFormat = {answerStyle: 'step'};
+    generatedAnswer.rephrase(responseFormat);
+    expect(updateResponseFormat).toHaveBeenCalledWith(responseFormat);
+  });
+
+  it('initialize the format', () => {
+    const responseFormat: GeneratedResponseFormat = {answerStyle: 'concise'};
+    createGeneratedAnswer({
+      initialState: {responseFormat},
+    });
+
+    expect(updateResponseFormat).toHaveBeenCalledWith(responseFormat);
+  });
+
+  it('dispatches a like action', () => {
```

Comment:
I don't think we necessarily need to retest everything that the core controller does here, since it's already tested in headless-core-generated-answer.test.ts!

you could simply test, the `state` getter, and the `rephrase`, `retry`, and `reset` methods.